runsc/boot/filter/config.go

```diff
@@ -247,12 +247,14 @@ var allowedSyscalls = seccomp.SyscallRules{
 		{seccomp.AllowAny{}, seccomp.AllowValue(syscall.SHUT_RDWR)},
 	},
 	syscall.SYS_SIGALTSTACK:     {},
+	syscall.SYS_SENDFILE:        {},
```

Comment:
I do not think we should add these syscalls to the filters. These are getting us into more fringe syscalls which we don't really need.

If I understand correctly, we wouldn't actually expect to get much use out of this. The common users of sendfile(2) (as I understand it) are web servers serving static files. In gVisor, those servers would be passing a netstack epsocket FD to sendfile and would thus not be eligible for the host optimization anyway.

Reply:
Yeah, I had debated this. What if this supports socket endpoints only, and we allow sendfile with network passthrough? That's the only case when this would be called.

Reply:
Sounds reasonable to me, network passthrough explicitly trades security for performance.

Reply:
Hm, I would be much more comfortable with that. However, that is still a real mode that users may use that we should keep as secure as possible. It can't just be the dumping ground for any syscall we want.